Mathlib/Tactic/MinImports.lean

```diff
@@ -59,34 +132,32 @@ def getAllImports {m : Type → Type} [Monad m] [MonadResolveName m] [MonadEnv m
   for t1 in ts do
     let tns := t1::(← resolveGlobalName t1).map Prod.fst
     for t in tns do
-      let new := ← match env.getModuleIdxFor? t with
-        | some t => return (hm.find? t).get!
-        | none   => return default --getMainModule
-        if !fins.contains new then fins := fins.insert new
-  return fins
-
-/-- `getIrredundantImports cmd` takes a `Syntax` input `cmd`.
-It returns the `NameSet` consisting of a minimal collection of module names whose transitive
+      let new := match env.getModuleIdxFor? t with
+        | some t => (hm.find? t).get!
+        | none   => .anonymous -- instead of `getMainModule`, we omit the current module
+      if !fins.contains new then fins := fins.insert new
+  return fins.erase .anonymous
+
+/-- `getIrredundantImports env importNames` takes an `Environment` and a `NameSet` as inputs.
```

Comment:
Is this functionality not already provided in the `import-graph` repo?

If so, please use, if not, ideally eventually this material could move there, but no hurry.

Reply:
The closest that I could find in `ImportGraph` is `Lean.Environment.findRedundantImports`, which is the function that I am using in `env.findRedundantImports`.  This is indeed really close: it determines precisely the imports that I want to exclude!  I could not find the command that I defined, since `ImportGraph` proceeds to define some `down`/`upstream` of, containing all imports above or below and, for `#min_imports`, the commands are already mixed up with the `Expr` information with all the declarations in the environment.

I am happy to move this and most of the new `#min_imports` functionality to `ImportGraph`, but I would rather do that later, after testing it in mathlib: I imagine that quite a few improvements will be necessary and doing it within mathlib should have a quicker turnaround and involve fewer inter-repo messing around.